rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -351,9 +359,10 @@ protected LRAService getLraService() {
     /**
      * return the current status of the LRA
      *
-     * @return null if the LRA is still active (not closing, cancelling, closed or cancelled
+     * @return current LRA status, will not return <code>null</code>
      */
     public LRAStatus getLRAStatus() {
+        assert status != null;
```

Comment:
Agreed, status can never be null. But why do we need to be asserting that.

Reply:
assertion is good to demonstrate that the code is designed in way where the `status` won't be null. The developer says that he is aware of that fact and announce it to anybody reading the code later.

Reply:
If we assert program invariant's everywhere they exist then the code would be littered with assertions making the program difficult to read.

Add a comment to the field if you really want to stress that the status can only take values according to a well defined state model

Reply:
agree, I think these are superfluous. I will change this.